app/src/main/java/com/github/h3lp3rs/h3lp/SignInActivity.kt

```diff
@@ -0,0 +1,116 @@
+import android.widget.ImageButton
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.GoogleAuthProvider
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.ktx.Firebase
+
+class SignInActivity : AppCompatActivity() {
+
+    private lateinit var googleSignInClient: GoogleSignInClient
+    private lateinit var auth: FirebaseAuth
+    private val serverClientId = "899579782202-t3orsbp6aov3i91c99r72kc854og8jad.apps.googleusercontent.com"
+
+    override fun onStart() {
+        super.onStart()
+
+        check()
+    }
+
+    /**
+     * Check if the current user is already signed in and update activity accordingly
+     */
+    private fun check() {
+        val currentUser = auth.currentUser
+        if (currentUser != null) {
+            val intent = Intent(this, HomeActivity::class.java)
+            startActivity(intent)
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_sign_in)
+
+        // Initialize Firebase Auth
+        auth = Firebase.auth
+
+        makeRequest()
+
+        findViewById<ImageButton>(R.id.signInButton).setOnClickListener{ signIn() }
+    }
+
+    /**
+     * Configure Google Sign In and build a Google sign in client with the options specified
+     */
+    private fun makeRequest() {
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestIdToken(serverClientId)
+            .requestEmail()
+            .build()
+
+        googleSignInClient = GoogleSignIn.getClient(this, gso)
+    }
+
+    /**
+     * Launch the Google sign in request
+     */
+    private fun signIn() {
+        val signInIntent = googleSignInClient.signInIntent
+        resultLauncher.launch(signInIntent)
+    }
+
+    // Handle sign in request result
+    private val resultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()){
+        result ->
+            if(result.resultCode == Activity.RESULT_OK){
+                val task = GoogleSignIn.getSignedInAccountFromIntent(result.data)
+                try {
+                    // Google Sign In was successful, authenticate with Firebase
+                    val account = task.getResult(ApiException::class.java)!!
+                    Log.d(TAG, "firebaseAuthWithGoogle:" + account.id)
+                    firebaseAuthWithGoogle(account.idToken!!)
+                } catch (e: ApiException) {
+                    // Google Sign In failed, display a message to the user.
+                    Log.w(TAG, "Google sign in failed", e)
```

Comment:
Put message in res/string.xml for maintenance and future translation

Reply:
the logs were only for debug, I will remove them
Thanks!